criu/ipc_ns.c

```diff
@@ -790,26 +790,20 @@ static int prepare_ipc_msg(int pid)
 static int restore_content(void *data, struct cr_img *img, const IpcShmEntry *shm)
 {
 	int ifd;
-	ssize_t size, off;
+	ssize_t size, ret;
 
 	ifd = img_raw_fd(img);
 	if (ifd < 0) {
 		pr_err("Failed getting raw image fd\n");
 		return -1;
 	}
-	size = round_up(shm->size, sizeof(u32));
-	off = 0;
-	do {
-		ssize_t ret;
-
-		ret = read(ifd, data + off, size - off);
-		if (ret <= 0) {
-			pr_perror("Failed to write IPC shared memory data");
-			return (int)ret;
-		}
 
-		off += ret;
-	} while (off < size);
+	size = round_up(shm->size, sizeof(u32));
+	ret = read_all(ifd, data, size);
+	if (ret != size) {
+		pr_perror("Failed to write IPC shared memory data");
+		return (int)ret;
```

Comment:
```suggestion
		return -1;
```
As the caller(s) do not know the total size so they compare only to `-1`. Its caller is only `prepare_ipc_shm_pages` and its only caller `prepare_ipc_shm_seg` uses for error checking:
```
        ret = prepare_ipc_shm_pages(img, shm);
        if (ret < 0) {
```